### PR TITLE
Don't try to compile brave-rewards and brave-sync when disabled 

### PR DESCRIPTION
**Pull request by akshay24495**
Fixes https://github.com/brave/brave-browser/issues/8072

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [x] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by akshay24495** [comment]
I could not link the issue to the pull request ( https://github.com/brave/brave-browser/issues/8072#issue-560308876 )

**Comment by bsclifton** [comment]
@bridiver @NejcZdovc @darkdh can you check this one out please? 😄 Changes LGTM

**Comment by NejcZdovc** [comment]
@bsclifton this PR needs to be moved in our repo so that we can run CI on it

**Comment by bsclifton** [comment]
Created https://github.com/brave/brave-core/pull/4720 so that CI can run 😄

**Comment by bsclifton** [comment]
Successful CI run with https://github.com/brave/brave-core/pull/4720 🎉 

Ready for review @NejcZdovc

**Comment by akshay24495** [comment]
May I know what is continuous integration and how to fix it?

**Comment by akshay24495** [comment]
@NejcZdovc Kindly review

**Comment by akshay24495** [comment]
@bsclifton Any updates?

File: chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc [hunk 1]
```diff
@@ -11,17 +11,27 @@
...
 #include "brave/components/brave_sync/grit/brave_sync_resources.h"
+#endif
 #include "brave/components/brave_webtorrent/grit/brave_webtorrent_resources.h"
 
 namespace extensions {
 
   bool IsComponentExtensionWhitelisted(const std::string& extension_id) {
     const char* const kAllowed[] = {
       brave_extension_id,
+      #if BUILDFLAG(BRAVE_REWARDS_ENABLED)
```
Comment (bridiver) [inline chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc]: preprocessor directives should not be indented

File: chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc [hunk 2]
```diff
@@ -11,17 +11,27 @@
 #include "brave/common/extensions/extension_constants.h"
 #include "brave/components/brave_extension/grit/brave_extension.h"
 #include "components/grit/brave_components_resources.h"
+#include "brave/components/brave_rewards/browser/buildflags/buildflags.h"
```
Comment (bridiver) [inline chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc]: deps for these buidflags need to be added to brave/browser/BUILD.gn "browser_process" so they get included for chromium_src

File: chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc (hunk 1 above)
Comment (akshay24495) [reply chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc]: I have removed the indentation for preprocessor.

File: chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc (hunk 2 above)
Comment (akshay24495) [reply chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc]: I have added the buildflags in deps of browser_process in brave/browser/BUILD.gn

Comment (bridiver) [inline chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc]: This is likely failing CI as a lint error. Includes need to be alphabetized

File: chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc [hunk 3]
```diff
@@ -11,17 +11,27 @@
...
+#include "brave/components/brave_sync/buildflags/buildflags.h"
+#if BUILDFLAG(BRAVE_REWARDS_ENABLED)
 #include "brave/components/brave_rewards/resources/extension/grit/brave_rewards_extension_resources.h"
+#endif
+#if BUILDFLAG(ENABLE_BRAVE_SYNC)
 #include "brave/components/brave_sync/grit/brave_sync_resources.h"
+#endif
 #include "brave/components/brave_webtorrent/grit/brave_webtorrent_resources.h"
```
Comment (bridiver) [inline chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc]: this should be moved above the conditionally added includes

File: chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc (hunk 2 above)
Comment (bridiver) [reply chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc]: basically all the non-conditional (no `#if`) includes should be grouped an alphabetized and then all the conditional includes should be listed after (also alphabetized in each subgroup if there is more than one. It's usually best to leave a blank line between groups as well. Sorry I missed this the first time, but lint will definitely complain about it

File: chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc (hunk 3 above)
Comment (akshay24495) [reply chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc]: I have moved the conditionally added includes

File: chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc (hunk 2 above)
Comment (akshay24495) [reply chromium_src/chrome/browser/extensions/component_extensions_whitelist/whitelist.cc]: Alphabetized the non-conditional includes

**Comment by bsclifton** [comment]
Created https://github.com/brave/brave-core/pull/5124 so that we can let this run through our CI system 😄 @akshay24495 can you squash the commits up into one?

**Comment by bsclifton** [comment]
CI passed with https://github.com/brave/brave-core/pull/5124 🎉 I can help you squash these down and will ping @bridiver for re-approval

File: browser/BUILD.gn [hunk 4]
```diff
@@ -136,6 +136,8 @@ source_set("browser_process") {
     "//brave/components/brave_webtorrent/browser/buildflags",
     "//brave/components/content_settings/core/browser",
     "//brave/components/greaselion/browser/buildflags",
+    "//brave/components/brave_rewards/browser/buildflags",
```
Comment (bridiver) [inline browser/BUILD.gn:139]: sorry last thing, I know a bunch of these deps are already out of alphabetical order, but can you alphabetize them relative to this group of deps at least? https://github.com/brave/brave-core/pull/4513/files#diff-84f8700538180eec90bb2f4c9dff4da4R132
so `"//brave/components/brave_rewards/browser/buildflags",` would go after `"//brave/components/brave_rewards/browser",` and `"//brave/components/brave_sync/buildflags",` would go after `"//brave/components/brave_shields/common",`

Comment (bridiver) [reply browser/BUILD.gn:139]: at least that way we won't be adding to the problem

Comment (akshay24495) [reply browser/BUILD.gn]: Changes have been done

**Pull request closed without merging**